collector/src/descriptor/kind/utils.rs

```diff
@@ -138,4 +138,15 @@ macro_rules! extract_desc {
     };
 }
 
+pub(crate) fn hashmap_from_kv_vec(data: Vec<&str>) -> HashMap<String, String> {
+    let res: HashMap<String, String> = data.iter().fold(HashMap::new(), |mut res, val| {
+        if let Some(t) = val.split_once('=') {
+            res.entry(t.0.to_owned()).or_insert_with(|| t.1.to_owned());
+            res
+        } else {
+            res
+        }
+    });
```

Comment:
could you rewrite this as a (flat_)map + collect?

I'm not sure about the semantic, shouldn't an entry with no `=` be considered invalid instead of being skipped over?